### PR TITLE
Task 2

### PR DESCRIPTION
**Pull request by aahinski**
Implemented product search and sorting, added PDP and product price history.

### CODE REVIEW

File: src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java [hunk 1]
```diff
@@ -1,81 +1,136 @@
...
+            } else {
+                /* sorting by default */
+                desiredProducts = sortBySortOrderAndSortField(unsortedDesiredProductsStream,
+                        SortField.price, SortOrder.desc);
+            }
+        } else {
+            desiredProducts = sortBySortOrderAndSortField(unsortedDesiredProductsStream, 
+                    sortField, sortOrder);
+        }
+
```
Comment (AnnaKrikun) [inline src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java]: Can methods sortByRelevance and sortBySortOrderAndSortField return Comparator which then can be used in return statement? You can build comparator at first:
         Comparator<Product> comparator = constructComparator(query, softField, sortOrder);
and then filter and sort results

File: src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java [hunk 2]
```diff
@@ -1,81 +1,136 @@
...
-
-            products.remove(nonUpdatedProduct);
-            products.add(product);
-        } else {
+        try {
+            Product productToUpdate = getProduct(product.getId());
+            product.setPriceHistoryList(productToUpdate.getPriceHistoryList());
+            product.getPriceHistoryList()
+                    .add(new PriceHistory(LocalDateTime.now(), product.getPrice(), product.getCurrency()));
+            product.setId(productToUpdate.getId());
```
Comment (AnnaKrikun) [inline src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java]: Why is it needed?

File: src/main/webapp/WEB-INF/pages/errorProductNotFound.jsp [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<%@ page contentType="text/html;charset=UTF-8" %>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<%@ taglib prefix="tags" tagdir="/WEB-INF/tags" %>
+
+<tags:master pageTitle="Product not found">
+    <h1>Product not found =(</h1>
```
Comment (AnnaKrikun) [inline src/main/webapp/WEB-INF/pages/errorProductNotFound.jsp]: Could you add id of the product which was not found?

**Comment by AnnaKrikun** [comment]
1. Add footer
2. Show price history in popup not in separate page

File: src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java (hunk 2 above)
Comment (aahinski) [reply src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java]: Line 122 is not needed, but the rest of the code is updating the price history: the updated product now has all its fields updated and new item in the price history (in future commits I will handle the case when the price is remaining the same).

**Comment by aahinski** [comment]
Task 2: implemented PDP, PLP, Product price-history

**Pull request merged**